bavi/modules/url.py

```diff
@@ -0,0 +1,108 @@
+
+IGNORE_REGEX = r'(?i)\.(jpg|jpg:large|png|gif|bmp|tiff|psd|mp4|mkv|avi|mov|mpg|vob|mp3|aac|wav|flac|ogg|mka|wma|zip|rar|7z|tar|iso|exe|dll|cfg|rc|reg|pxe|efi)$'
+
+compiled_url_regex = re.compile(URL_REGEX, re.UNICODE)
+compiled_ip_regex = re.compile(IP_REGEX)
+compiled_ignore_regex = re.compile(IGNORE_REGEX)
+
+TITLE_FORMAT = '[ {0} ] - {1}'
+
+
+def init(bot):
+    bot.add_command('title', title_command)
+    bot.add_matcher(compiled_url_regex, title_command, priority='low')
+    bot.add_matcher(compiled_ip_regex, title_command, priority='low')
+
+
+def title_command(bot, source, target, message, **kwargs):
+    log.debug('url module triggered on: ' + message)
+
+    if len(message) > 0:
+        valid_urls = get_valid_urls(message)
+
+        if len(valid_urls) > 0:
+            #limiting to a maximum of 5 URLs in one request
+            for url in valid_urls[:5]:
+                response = get_user_readable_response(url)
+
+                if len(response) > 0:
+                    bot.say(target, response)
+
+
+def get_user_readable_response(url):
+    response = ''
+
+    if len(url) > 0:
+        #now start a request for the page title
+        title = get_http_title(url)
+
+        if (len(title) > 0):
+            response = TITLE_FORMAT.format(title, get_domain_from_url(url))
+
+    return response
+
+
+def get_valid_urls(string_statement):
+    valid_matches = []
+    valid_urls = []
+
+    valid_matches += re.findall(compiled_url_regex, string_statement)
+    valid_matches += re.findall(compiled_ip_regex, string_statement)
+
+    log.debug(valid_matches)
+
+    for match in valid_matches:
+        if compiled_ignore_regex.search(match[0]) is None:
+            valid_urls.append(match[0])
+
+    log.debug('Found: ' + ', '.join(str(url) for url in valid_urls))
+
+    return valid_urls
+
+
+def get_http_title(url):
+    title = ''
+
+    try:
+        log.debug('Requesting url %s', url)
+
+        #TODO: Possibly bring these parameters to user defined setting.
+        #Currently set to first 512KBytes from server and timeout at 5s
+        headers = {"Range": "bytes=0-512000"}
+        
+        #use `with` to make sure the connection goes back into the pool
+        with requests.get(url, headers=headers, timeout=5) as r:
+            log.debug('%s responded with %d', url, r.status_code)
+            
+            if r.headers['content-type'].startswith('text/html'):
+                bs = BeautifulSoup(r.content[1:512000], 'html.parser')
+                title = bs.find('title').string
+
+    except Exception as e:
+        exception_msg = type(e).__name__ + ': ' + str(e)
+        log.exception('Could not process: %s', url)
+        self.say(event.target, exception_msg)
```

Comment:
I think you meant `bot.say` because `self` is undefined here.  I would also accept just getting rid of this because from my reading of the code above, an exception in this block would be a bug in the code and not just a faulty URL, since the URL has been validated before this method.